src/engine.cc

```diff
@@ -339,7 +339,10 @@ void EngineLoop::CmdIsReady() {
 
 void EngineLoop::CmdSetOption(const std::string& name, const std::string& value,
                               const std::string& context) {
-  options_.SetOption(name, value, context);
+  if (!options_.SetOption(name, value, context)) {
+    throw Exception("No such option: " + name);
```

Comment:
There is another similar PR: https://github.com/LeelaChessZero/lc0/pull/246
which returns false for SetOption on bad parameter value.
I think we at least have to have a plan how to support both. (Here if SetOption returns false, it will be not clear whether it's bad name or bad value).
I like throwing exceptions directly from inside of SetOption(), not sure if others consider that ugly. :)